src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DIPS.Xamarin.UI.Internal.Xaml;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    /// <summary>
+    ///     Toast control that would appear on top of the presented view
+    /// </summary>
+    public class Toast : BindableObject
+    {
+        private Toast()
+        {
+            Application.Current.PageAppearing -= OnPageAppearing;
+            Application.Current.PageAppearing += OnPageAppearing;
+
+            Application.Current.PageDisappearing -= OnPageDisappearing;
+            Application.Current.PageDisappearing += OnPageDisappearing;
+        }
+
+        /// <summary>
+        ///     Get the current instance of the Toast control
+        /// </summary>
+        public static Toast Current { get; } = new Toast();
+
+        private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
+        private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
+        private ToastLayout ToastLayout { get; set; }
+        private ToastOptions ToastOptions { get; set; }
+
+        private void OnPageDisappearing(object sender, Page page)
+        {
+            if (page is ContentPage contentPage)
+            {
+                CancellationSource.Cancel();
+                _ = CloseToast(contentPage);
+            }
+        }
+
+        private void OnPageAppearing(object sender, Page e)
+        {
+            Initialize();
+        }
+
+        /// <summary>
+        ///     Initialize Toast control.
+        /// </summary>
+        public void Initialize()
+        {
+            _ = GetToastContainerSettingUpIfNeededAsync();
+        }
+
+        /// <summary>
+        ///     Displays the Toast control
+        /// </summary>
+        /// <param name="text">Text to be displayed in the Toast control</param>
+        /// <param name="options">An <see cref="Action{ToastOptions}" /> to modify Toast options</param>
+        /// <param name="layout">An <see cref="Action{ToastLayout}" /> to modify Toast layout</param>
+        /// <returns>A void <c>Task</c></returns>
+        public async Task DisplayToastAsync(string text, Action<ToastOptions> options, Action<ToastLayout> layout)
```

Comment:
No need for the `Async` postfix, it's a `Task` so it's self-explanatory that its awaitable. 
We have skipped the `Async` postfix for other APIs in DXUI.

Reply:
If it's skipped in DXUI, then I can avoid it. However thought it returns a Task, in code it's not going to be so obvious if someone had used it in a non async method. Easier to spot with the postfix.

Reply:
Removed the popstfix